src/components/reset-password/tests/reset-password-2fa-before-integration.test.ts

```diff
@@ -0,0 +1,224 @@
+import request from "supertest";
+import { describe } from "mocha";
+import { expect, sinon } from "../../../../test/utils/test-utils";
+import nock = require("nock");
+import * as cheerio from "cheerio";
+import { PATH_NAMES } from "../../../app.constants";
+import decache from "decache";
+
+describe("Integration::reset password (in 2FA Before Reset Password flow)", () => {
```

Comment:
We need two test files because the environment variables cannot be modified once the integration test start, as they are set at the begging when creating the mock app. Thus we need to test that the functionalities work as expected with the switch turned on or off.

Reply:
I left all of the tests in the new file thinking that when the feature flag is turned on we can just delete the old integration test file, but, if needed, I can just leave the test where we check if it redirects to /auth-code.

Reply:
Got you, I think we could probably get around that by declaring a `let` variable on e.g. line 14, setting the value of that in the individual tests, and then using that variable to set the env variable on line 18, which would let us vary whether the env variable was on or not in the tests, but I'll let you decide what you want to do here.

Reply:
Ah got you, these integration tests are a bit annoyingly fiddly! Makes sense